src/conventions/h-group/clue-finder/determine-clue.js

```diff
@@ -44,55 +51,66 @@ export function evaluate_clue(game, action, clue, target, target_card, bad_touch
 	/** @type {string} */
 	let reason;
 
-	for (const { order, clued } of state.hands[target]) {
-		const card = hypo_game.common.thoughts[order];
-		const visible_card = state.deck[order];
+	const get_finessed_cards = (game) => {
+		return game.state.hands[action.giver].filter(c => !game.common.thoughts[c.order].clued && game.common.thoughts[c.order].finessed);
+	};
+
+	const finessed_before_clue = get_finessed_cards(game);
+	const finessed_after_clue = get_finessed_cards(hypo_game);
+	const lost_finesse = finessed_before_clue.filter(c => finessed_after_clue.find(other => other.order == c.order) === undefined);
+	if (lost_finesse.length > 0) {
+		reason = `cards ${lost_finesse.map(c => logCard(state.deck[c.order])).join(', ')} lost finesse`;
+	} else {
+		for (const { order, clued } of state.hands[target]) {
```

Comment:
One thing I was thinking while writing this PR is that it might be valuable to look for wrong inferences on all hands rather than just looking at the target. Seems like it could only help to catch cases where the clue has indirect consequences due to either a connection through someone else's hand or assumed good touch or other things like this.

Reply:
Could come up with an example where this would occur? I think if it affected other cards the focus wouldn't end up with the correct interpretation, but I'm open to ideas. There was a brief period where cluing a trash card "stole" the good touch inference from another card, but that was a bug in good touch elimination and not clue interpretation.

Reply:
My thinking was this could be a different way of detecting that we lost a finesse without needing to explicitly check that we lost a finesse - because some other cards would get incorrect inferences. E.g. if Alice gives a clue that loses an expected finese, then it also probably resulted in adding a connection on someone else who might think they have the connecting card?

Reply:
My plan wasn't to pursue this for this PR just thinking out loud (in comments :relaxed: )

Reply:
> E.g. if Alice gives a clue that loses an expected finese, then it also probably resulted in adding a connection on someone else who might think they have the connecting card?

Hmm, I'm not sure. I don't think it will try to connect on cards that clearly don't match, it might just think some kind of mistake has occurred. We could try it out, but based on the bot's current capabilities I think the benefit will be minimal, if any.